clients/web/src/lib/components/atoms/form/FileInput/FileInput.svelte

```diff
@@ -0,0 +1,273 @@
+    }
+
+    export let label: string;
+    export let value: FileInputValue<Multi> | undefined = undefined;
+    export let multiple: Multi = false as Multi;
+    export let showDropzone: FileInputShowDropzone = "when-dropping";
+
+    export let size: FormControlSize = "md";
+    export let placeholder: string | undefined = undefined;
+    export let disabled: boolean = false;
+    export let state: FormControlState = "none";
+    export let error: string | undefined = undefined;
+
+    const labelId = `file-input_${nanoid()}`;
+    const errorId = `file-input-error_${nanoid()}`;
+
+    // Internal state
+    let hasValue: boolean;
+    let inputButtonText: string;
+    let inputStatusText: string;
+    let dropzoneStatusText: string;
+    let render: "input" | "dropzone";
+
+    // State passed to slots
+    let rejectedFiles: RejectedFile[];
+    let isDropping = false;
+
+    $: inputButtonText = multiple ? "Choose Files" : "Choose File";
+    $: hasValue = (!multiple && value !== undefined) || (multiple && (value as FileInputValue<true>)?.length > 0);
+
+    // Decide what to render based on `showDropzone` and `isDropping`
+    $: {
+        if (showDropzone === "never") {
+            render = "input";
+        } else if (showDropzone === "always") {
+            render = "dropzone";
+        } else {
+            render = isDropping ? "dropzone" : "input";
+        }
+    }
+
+    // Determine status text based on `multiple` and `value`
+    $: {
+        if (hasValue) {
+            if (multiple) {
+                const numFiles = (value as FileInputValue<true>).length;
+                inputStatusText = `${numFiles} files uploaded`;
+                dropzoneStatusText = `${numFiles} files uploaded`;
+            } else {
+                const filename = (value as FileInputValue<false>).name;
+                inputStatusText = filename;
+                dropzoneStatusText = filename;
+            }
+        } else {
+            inputStatusText = placeholder || "No file chosen";
+            dropzoneStatusText = multiple ? "Drop files here" : "Drop file here";
+        }
+    }
+
+    const onFiledrop = (e: CustomEvent<FileDropSelectEvent>): void => {
+        const {accepted, rejected} = e.detail.files;
+        console.log({accepted});
+        value = (multiple ? accepted : accepted?.[0]) as FileInputValue<Multi>;
+        rejectedFiles = rejected;
+        isDropping = false;
+    };
+</script>
+
+<div class="size-{size} state-{state}">
+    <label for={labelId}>{label}</label>
+
+    <div
+        class="filedrop"
+        class:disabled
+        use:filedrop={{multiple, disabled}}
+        on:filedrop={onFiledrop}
+        on:filedragenter={() => (isDropping = true)}
+        on:filedragleave={() => (isDropping = false)}
+    >
+        <!-- Hidden input only for screenreaders -->
+        <input id={labelId} type="file" aria-describedby={error ? errorId : undefined} {disabled} />
+
+        {#if render === "input"}
```

Comment:
Move the condition out of the variable and into the markdown

Reply:
Logic is complex enough that it warrants keeping it in `<script>`